includes/blocks/class-sensei-lesson-actions-block.php

```diff
@@ -37,6 +37,16 @@ public function __construct() {
 	 * @return string The block HTML.
 	 */
 	public function render( array $attributes, string $content ) : string {
+		$lesson = get_post();
+
+		if ( empty( $lesson ) ) {
+			return '';
+		}
+
+		if ( ! Sensei_Lesson::should_show_lesson_actions( $lesson->ID ) ) {
```

Comment:
I took most of the checks from the old buttons. If we wan't to update/streamline the rules of when each button is going to be displayed, this PR is a good chance to do it.